source/server/listener_impl.cc

```diff
@@ -123,15 +123,85 @@ Network::SocketSharedPtr ListenSocketFactoryImpl::getListenSocket() {
   return createListenSocketAndApplyOptions();
 }
 
+ListenerFactoryContextBaseImpl::ListenerFactoryContextBaseImpl(
+    Envoy::Server::Instance& server, ProtobufMessage::ValidationVisitor& validation_visitor,
+    const envoy::config::listener::v3::Listener& config, DrainManagerPtr drain_manager)
+    : server_(server), metadata_(config.metadata()), direction_(config.traffic_direction()),
+      global_scope_(server.stats().createScope("")),
+      listener_scope_(server_.stats().createScope(fmt::format(
+          "listener.{}.", Network::Address::resolveProtoAddress(config.address())->asString()))),
+      validation_visitor_(validation_visitor), drain_manager_(std::move(drain_manager)) {}
+
+AccessLog::AccessLogManager& ListenerFactoryContextBaseImpl::accessLogManager() {
+  return server_.accessLogManager();
+}
+Upstream::ClusterManager& ListenerFactoryContextBaseImpl::clusterManager() {
+  return server_.clusterManager();
+}
+Event::Dispatcher& ListenerFactoryContextBaseImpl::dispatcher() { return server_.dispatcher(); }
+Grpc::Context& ListenerFactoryContextBaseImpl::grpcContext() { return server_.grpcContext(); }
+bool ListenerFactoryContextBaseImpl::healthCheckFailed() { return server_.healthCheckFailed(); }
+Http::Context& ListenerFactoryContextBaseImpl::httpContext() { return server_.httpContext(); }
+const LocalInfo::LocalInfo& ListenerFactoryContextBaseImpl::localInfo() const {
+  return server_.localInfo();
+}
+Envoy::Runtime::RandomGenerator& ListenerFactoryContextBaseImpl::random() {
+  return server_.random();
+}
+Envoy::Runtime::Loader& ListenerFactoryContextBaseImpl::runtime() { return server_.runtime(); }
+Stats::Scope& ListenerFactoryContextBaseImpl::scope() { return *global_scope_; }
+Singleton::Manager& ListenerFactoryContextBaseImpl::singletonManager() {
+  return server_.singletonManager();
+}
+OverloadManager& ListenerFactoryContextBaseImpl::overloadManager() {
+  return server_.overloadManager();
+}
+ThreadLocal::Instance& ListenerFactoryContextBaseImpl::threadLocal() {
+  return server_.threadLocal();
+}
+Admin& ListenerFactoryContextBaseImpl::admin() { return server_.admin(); }
+// TODO(lambdai): Consider moving away from factory context since listener filter and network filter
+// have independent factory context.
```

Comment:
It's unclear where this TODO actually applies or what it means. Can you remove this or put this next to class docs somewhere?

Reply:
Moved there and list listenerMetadata as an example